From 992f44fab04a56bc4cbd003806b88affac1b031f Mon Sep 17 00:00:00 2001
From: Mirage Su <mirage.su@mic.com.tw>
Date: Fri, 1 Mar 2019 14:51:17 +0800
Subject: [PATCH] [MCT Common] Bug #10583 The Firmware Revision from 'ipmitool
 mc info' may be '0.00'

Return the CC_D5h while the software redundancy interface isn't created yet.
---
 apphandler.cpp      | 5 +++++
 include/ipmid/api.h | 1 +
 2 files changed, 6 insertions(+)
 mode change 100644 => 100755 include/ipmid/api.h

diff --git a/apphandler.cpp b/apphandler.cpp
index 72bbc6b..d7763de 100755
--- a/apphandler.cpp
+++ b/apphandler.cpp
@@ -752,6 +752,11 @@ ipmi_ret_t ipmi_app_get_device_id(ipmi_netfn_t netfn, ipmi_cmd_t cmd,
             dev_id.fw[1] = rev.minor % 10 + (rev.minor / 10) * 16;
             std::memcpy(&dev_id.aux, rev.d, 4);
         }
+        else{
+            //MiTAC Henbin Oem addition.
+            rc = IPMI_CC_NOT_SUP_IN_CUR_STAT;
+            return rc;
+        }
 
         // IPMI Spec version 2.0
         dev_id.ipmi_ver = 2;
diff --git a/include/ipmid/api.h b/include/ipmid/api.h
old mode 100644
new mode 100755
index b1c078b..2895fad
--- a/include/ipmid/api.h
+++ b/include/ipmid/api.h
@@ -128,6 +128,7 @@ enum ipmi_return_codes
     IPMI_CC_ILLEGAL_COMMAND = 0xCD,
     IPMI_CC_RESPONSE_ERROR = 0xCE,
     IPMI_CC_INSUFFICIENT_PRIVILEGE = 0xD4,
+    IPMI_CC_NOT_SUP_IN_CUR_STAT = 0xD5, //MiTAC Henbin OEM add
     IPMI_CC_UNSPECIFIED_ERROR = 0xFF,
 };
 
